### PR TITLE
Code formatting and linter

### PR DESCRIPTION
**Pull request by marciogranzotto**
As discussed on #47 this project adds `ktlint` to the project.
It also auto-formats and solves the linter warnings

Full list of changes:
- Add ktlint to the project
- Auto format and fix linter issues
- After a `ktlint --android --apply-to-idea-project` command, changed the `gitignore` file to include styling idea files. That way any auto-formats that other contributor do after this, will be on the ktlint style.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -22,4 +22,5 @@ cache:
 
 script:
   - chmod ugo+x .travis/script.sh
-  - ./.travis/script.sh
\ No newline at end of file
+  - ./.travis/script.sh
+  - ./gradlew ktlintCheck
```
Comment (CedrickFlocon) [inline .travis.yml]: You can put the task in the script.sh just after the lint command

File: build.gradle [hunk 2]
```diff
@@ -6,24 +6,31 @@ buildscript {
...
 
 task clean(type: Delete) {
     delete rootProject.buildDir
 }
+
+ktlint {
+    android = true
+}
+
+apply plugin: "org.jlleitschuh.gradle.ktlint"
```
Comment (CedrickFlocon) [inline build.gradle]: Could you reorder the file like this :
buildscript
apply (file and plugin)
allprojects 
...

File: .gitignore [hunk 3]
```diff
@@ -1,8 +1,134 @@
-.DS_Store
+# Created by https://www.gitignore.io/api/androidstudio
```
Comment (CedrickFlocon) [inline .gitignore]: We should not use a template. A lot of stuff is not used.

File: .gitignore [hunk 4]
```diff
@@ -1,8 +1,134 @@
...
+
+# External native build folder generated in Android Studio 2.2 and later
+.externalNativeBuild
+
+# NDK
+obj/
+
+# IntelliJ IDEA
 *.iml
-.idea/
```
Comment (CedrickFlocon) [inline .gitignore]: commit IDE file is not a good practice. 
We will  link our project to an IDE and also a specific version.

Comment (marciogranzotto) [reply .gitignore]: ok, I reverted these changes

**Comment by marciogranzotto** [comment]
@CedrickFlocon 
- Reverted `gitignore` changes
- Made the requested changes
- Added a Quality section on README

**Pull request merged**